string shape
float32[] position
float32[] orientation
float32[] color
---